connectors/airwatch/src/main/java/com/vmware/connectors/airwatch/config/ManagedApp.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2017 VMware, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+package com.vmware.connectors.airwatch.config;
+
+import java.util.Objects;
+
+/**
+ * Created by harshas on 01/31/18.
+ */
+public class ManagedApp {
+
+    private String name;
+
+    private String id;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(name, id);
```

Comment:
Your hashCode and equals don't agree.

equals is only comparing id, but hashCode is comparing name and id.

The rule is for any 2 objects that are equal, they must have the same hashCode value.

When you break that rule, really bad things can happen with collections.  I can explain more and give you some examples in chat.

For now, either check name in your equals method also, or remove name from the hash.

It'd probably be better to just use Pojomatic (I'm not sure how you specify which properties to use, but I'm sure there is a way).

Reply:
Yeah, use Pojomatic.